pallets/liquid-staking/src/lib.rs

```diff
@@ -708,13 +737,22 @@ pub mod pallet {
                     return Err(Error::<T>::NotWithdrawn.into());
                 }
 
-                T::Assets::transfer(
-                    Self::staking_currency()?,
-                    &Self::account_id(),
-                    &who,
-                    amount,
-                    false,
-                )?;
+                if who == Self::loans_account_id() {
```

Comment:
claim_for `loans_account` is so special, maybe we can do it once `advance_era`. or in a hook function.

Reply:
that is my question also. who will invoke the `do_claim_for` ? from some offchain service or internal onchain hook?

Reply:
@yrong currently https://github.com/parallel-finance/stake-client will invoke it